src/types/value_ref.rs

```diff
@@ -87,8 +92,14 @@ impl ValueRef<'_> {
             ValueRef::Blob(_) => Type::Blob,
             ValueRef::Date32(_) => Type::Date32,
             ValueRef::Time64(..) => Type::Time64,
+            ValueRef::List(arr, _) => arr.data_type().into(),
         }
     }
+
+    /// Returns an owned version of this ValueRef
+    pub fn to_owned(&self) -> Value {
```

Comment:
Could instead using the standard `ToOwned` trait impl here.
```rust
impl ToOwned for ValueRef<'_> {
    type Owned = Value;

    fn to_owned(&self) -> Self::Owned {
      *self.into()
    }
}
```

Reply:
Unfortunately this conflicts with the default implementation provided by Clone, and would require reworking elsewhere